### PR TITLE
Grand flattening and reorganization of `tick` 

### PR DESCRIPTION
**Pull request by stephanegaiffas**
For issue #124

### CODE REVIEW

**Comment by stephanegaiffas** [comment]
I'm all set with the documentation... I think this is ready now !
@Mbompr @Dekken :  time for reviewing

**Comment by Dekken** [comment]
- update the sh/gtest* files so they search the right directory

- linting fails - but perhaps we can ignore it for the moment?

File: doc/conf.py [hunk 1]
```diff
@@ -92,9 +92,9 @@
 # built documents.
 #
 # The short X.Y version.
-version = '0.3'
+version = '0.4'
 # The full version, including alpha/beta/rc tags.
-release = '0.3.0'
+release = '0.4.0'
```
Comment (Mbompr) [inline doc/conf.py]: I like to change the doc version when we do the pip release and upgrade the version in setup.py as well.

File: doc/index.rst [hunk 2]
```diff
@@ -31,91 +27,185 @@
...
+  <!-- tick.hawkes.simulation and tick.plot -->
+
+  <div class="row">
      <div class="col-sm-4 col-sm-offset-1">
-        <a class="reference internal" href="modules/api.html">
-           <h2>API reference</h2>
+        <a class="reference internal" href="modules/simulation.html">
+           <h2>tick.hawkes.simulation</h2>
+        </a>
+        <p>Basic tools for simulation, such as simulation of model weights and
```
[the remark this answers is not in the document]
Comment (Mbompr) [reply doc/index.rst]: @stephanegaiffas  Was this caption for tick.simulation? Do you want to keep it in the doc?

[the remark this answers is not in the document]
Comment (stephanegaiffas) [reply doc/index.rst]: Yes I want to keep it :) I'll do the last changes... (Philip moved everything in tick.hawkes and forgot to correct this one, he's been a bad boy :)

**Pull request merged**